test/addons/argocd/kustomization.yaml

```diff
@@ -0,0 +1,80 @@
+# SPDX-FileCopyrightText: The RamenDR authors
+# SPDX-License-Identifier: Apache-2.0
+
+# yamllint disable rule:line-length
+---
+resources:
+  - https://raw.githubusercontent.com/argoproj/argo-cd/release-2.11/manifests/install.yaml
+
+patches:
+  - target:
+      kind: Role
+      name: argocd-applicationset-controller
+    patch: |-
```

Comment:
To mimic create application step in ACM, I create ApplicationSet with spec like below, spec/generators/clusterDecisionResource using a cm pointing to a placementdecision.
```
  spec:
    generators:
    - clusterDecisionResource:
        configMapRef: appset-deploymentrbd-busybox
        labelSelector:
          matchLabels:
            cluster.open-cluster-management.io/placement: appset-deploymentrbd-busybox
        requeueAfterSeconds: 180
```
and cm data as:
```
data:
  apiVersion: cluster.open-cluster-management.io/v1beta1
  kind: placementdecisions
  matchKey: clusterName
  statusListKey: decisions
```

Appset controller does not know which kind we will use, and since here we choose to use placementdecisions, appset controller needs additional role to list and read it. Otherwise the appset CR reports error:
```
status:
  conditions:
  - lastTransitionTime: "2024-06-04T09:37:04Z"
    message: 'placementdecisions.cluster.open-cluster-management.io is forbidden:
      User "system:serviceaccount:argocd:argocd-applicationset-controller" cannot
      list resource "placementdecisions" in API group "cluster.open-cluster-management.io"
      in the namespace "argocd"'
    reason: ApplicationGenerationFromParamsError
    status: "True"
    type: ErrorOccurred
```

Such role addition is also documented at https://github.com/argoproj/applicationset/blob/master/examples/clusterDecisionResource/README.md

Reply:
comment is added to the file.